packages/syft/src/syft/core/tensor/autodp/phi_tensor.py

```diff
@@ -1040,34 +1040,51 @@ def __gt__(self, other: SupportedChainType) -> Union[PhiTensor, GammaTensor]:
         else:
             raise NotImplementedError  # type: ignore
 
-    # Re enable after testing
-    # def dot(
-    #     self, other: Union[PhiTensor, GammaTensor, np.ndarray]
-    # ) -> Union[PhiTensor, GammaTensor]:
-    #     if isinstance(other, np.ndarray):
-    #         print("We here or what?")
-    #         return PhiTensor(
-    #             child=np.dot(self.child, other),
-    #             min_vals=np.dot(self.min_vals, other),
-    #             max_vals=np.dot(self.max_vals, other),
-    #             data_subjects=self.data_subjects,
-    #         )
-    #     elif isinstance(other, PhiTensor):
-    #         if (
-    #             len(self.data_subjects.one_hot_lookup) > 1
-    #             or len(other.data_subjects.one_hot_lookup) > 1
-    #         ):
-    #             return self.gamma.dot(other.gamma)
-    #         elif (
-    #             len(self.data_subjects.one_hot_lookup) == 1
-    #             and len(other.data_subjects.one_hot_lookup) == 1
-    #             and self.data_subjects.one_hot_lookup != other.data_subjects.one_hot_lookup
-    #         ):
-    #             return self.gamma.dot(other.gamma)
-    #     elif isinstance(other, GammaTensor):
-    #         return self.gamma.dot(other)
-    #     else:
-    #         raise NotImplementedError
+# Re enable after testing
+    def dot(self, other: Union[PhiTensor, GammaTensor, np.ndarray]) -> Union[PhiTensor, GammaTensor]:
+        if isinstance(other, np.ndarray):
+            # print("We here or what?")
+            return PhiTensor(
+                child=np.dot(self.child, other),
+                min_vals=np.dot(self.min_vals, other),
+                max_vals=np.dot(self.max_vals, other),
+                data_subjects=self.data_subjects,
+            )
+        elif isinstance(other, PhiTensor):
+            if (
+                len(self.data_subjects.one_hot_lookup) > 1
+                or len(other.data_subjects.one_hot_lookup) > 1
+            ):
+                return self.gamma.dot(other.gamma)
+            elif (
+                    len(self.data_subjects.one_hot_lookup) == 1
+                    and len(other.data_subjects.one_hot_lookup) == 1
+                    and self.data_subjects.one_hot_lookup
+                    != other.data_subjects.one_hot_lookup
+            ):
+                return self.gamma.dot(other.gamma)
+            elif (
+                    len(self.data_subjects.one_hot_lookup) == 1
+                    and len(other.data_subjects.one_hot_lookup) == 1
+                    and self.data_subjects.one_hot_lookup
+                    == other.data_subjects.one_hot_lookup
+                ):
+                return PhiTensor(
+                    child=np.dot(self.child.child, other.child.child),
```

Comment:
We could call dot function on the child, and we could implement np.dot for the below tensor chains
`child = self.child.dot(other.child)`